### PR TITLE
IPC Updates & Minor Bugfixes

### PR DESCRIPTION
**Pull request by corigne**
**Primary Changes:**
- Updated IPC method to align with with polybar version 3.6 IPC changes which require the use of polybar-msg.
Addresses #36 .
- The listener service can now detect Spotify on launch/reload of the service or if Spotify is launched while the service is running.

**Minor Changes:**
- Some additional debug statements were added for verbosity.
- Minor adjustments to a couple of debug statements for clarity.

**Bug-fixes:**
- Fixed a bug which prevented the module hooks from running until the track was changed at least once.
- Fixed a bug where only only some hooks were updated on property change. At launch, in combination with the other bug resulted in a partially rendered module until the user executed all 3 types of playback actions (Play/Pause, Next, Previous)

### CODE REVIEW

**Comment by k1ng440** [comment]
Hey @corigne, 

Thank you for this pull request.  I know it's not my business but changes of  [spotify-listener.c](https://github.com/mihirlad55/polybar-spotify-module/pull/38/files#diff-486a0b5409363858f292a66d3588ab858d83c5d43dfde642a87860cbeb42b632) is massive due to your formatting which adds unnecessary complexity to code review.

**Comment by corigne** [comment]
I didn't actually add much. Due to shifting things around the diff looks larger than it is, unfortunately. I could try to clean that up but I'm not sure where to start without rewriting history. I could create another branch and then append my changes rather than putting them in arbitrary locations.

However if the maintainer and others believe it would be better served to excise the two functions I wrote into a separate file then I can do that. 

Moving the two functions outside of the file could also unnecessarily break up relevant code in favor of reducing the size of the diff in my opinion that's not a great tradeoff.

Happy to try to make the diff less confusing when I have time again, though if that makes life easier for the maintainers.

Tl;dr I'll see what I can do about the diff size if it's a problem.

**Comment by corigne** [comment]
Closing. Will reopen when I've reduced the diff as much as possible for clarity.
